appcues/src/main/java/com/appcues/debugger/DebuggerViewModel.kt

```diff
@@ -151,30 +176,37 @@ internal class DebuggerViewModel(
     }
 
     fun onDragEnd() {
-        _uiState.value = Idle
+        _uiState.value = Idle(mode)
     }
 
     fun onDismiss() {
         _uiState.value = Dismissing
     }
 
     fun onFabClick() {
-        if (_uiState.value == Idle) {
-            _uiState.value = Expanded()
-        } else if (_uiState.value is Expanded) {
-            _uiState.value = Idle
+        when (val state = _uiState.value) {
+            is Idle -> {
+                when (state.mode) {
+                    is Debugger -> _uiState.value = Expanded(state.mode)
+                    is ScreenCapture -> Unit
```

Comment:
tap action is nothing right now - but this is where we'll carry on and add the capture logic and then the confirmation dialog.